src/components/Collapsible/Collapsible.tsx

```diff
@@ -30,165 +24,83 @@ export interface CollapsibleProps {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({
-        animationState: 'idle',
-      });
-
-      return;
+  useEffect(() => {
+    if (open !== isOpen) {
+      setAnimationState('measuring');
     }
-
-    requestAnimationFrame(() => {
-      const heightNode = this.heightNode.current;
-      switch (animationState) {
-        case 'idle':
-          break;
-        case 'measuring':
-          this.setState({
-            animationState: wasOpen ? 'closingStart' : 'openingStart',
-            height: wasOpen && heightNode ? heightNode.scrollHeight : 0,
-          });
-          break;
-        case 'closingStart':
-          this.setState({
-            animationState: 'closing',
-            height: 0,
-          });
-          break;
-        case 'openingStart':
-          this.setState({
-            animationState: 'opening',
-            height: heightNode ? heightNode.scrollHeight : 0,
-          });
-      }
-    });
-  }
-
-  render() {
-    const {id, expandOnPrint, open, children, transition} = this.props;
-    const {animationState, height} = this.state;
-    const parentCollapsibleExpanding = this.context;
-
-    const animating = animationState !== 'idle';
-
-    const wrapperClassName = classNames(
-      styles.Collapsible,
-      open && styles.open,
-      animating && styles.animating,
-      !animating && open && styles.fullyOpen,
-      expandOnPrint && styles.expandOnPrint,
-    );
-
-    const displayHeight = collapsibleHeight(open, animationState, height);
-
-    const content = animating || open || expandOnPrint ? children : null;
-
-    const transitionProperties = transition
-      ? {
-          transitionDuration: `${transition.duration}`,
-          transitionTimingFunction: `${transition.timingFunction}`,
-        }
-      : null;
-
-    return (
-      <ParentCollapsibleExpandingContext.Provider
-        value={
-          parentCollapsibleExpanding || (open && animationState !== 'idle')
-        }
-      >
-        <div
-          id={id}
-          aria-hidden={!open}
-          style={{
-            maxHeight: `${displayHeight}`,
-            ...transitionProperties,
-          }}
-          className={wrapperClassName}
-          ref={this.node}
-          onTransitionEnd={this.handleTransitionEnd}
-        >
-          <div ref={this.heightNode}>{content}</div>
-        </div>
-      </ParentCollapsibleExpandingContext.Provider>
-    );
-  }
-
-  private handleTransitionEnd = (event: TransitionEvent) => {
-    const {target} = event;
-    if (target === this.node.current) {
-      this.setState({animationState: 'idle', height: null});
+  }, [open, isOpen]);
+
+  // If collapsible defaults to open, set an initial height
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    if (!open || !collapisbleContainer.current) return;
+    setHeight(collapisbleContainer.current.scrollHeight);
+  });
```

Comment:
No dependency array here means it runs on every rerender. An empty array means only run on initial mount, is that what you want?

From https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
Thanks Ben! That makes sense, I always get these mixed up 🤦‍♂️